src/hooks/users.ts

```diff
@@ -105,6 +105,31 @@ export const useRecentWorldUsers = (): {
   );
 };
 
+export const useRecentWorldUser = (
+  userId: string
+): {
+  user?: WithId<User>;
+  userLastSeenIn?: string;
+} => {
+  const { recentWorldUsers } = useRecentWorldUsers();
+
+  const user = useMemo(() => {
+    return recentWorldUsers.find((worldUser) => worldUser.id === userId);
+  }, [userId, recentWorldUsers]);
+
+  const userLastSeenIn = Object.keys(user?.lastSeenIn ?? {})[0];
+
+  return { user, userLastSeenIn };
+};
```

Comment:
What's the point of having this hook, if you have `selectedUserProfile` at the level of calling it?

Reply:
Ah, I guess to check if the user is online and return his location. Well, it should be divided into two. First hook should check if user is online. 
NOTE: Also reuse it in `UserAvatar`

![image](https://user-images.githubusercontent.com/63194656/120630412-7fd4d280-c46f-11eb-9c24-b40ebcddd3bb.png)

Second util should be util function and return either undefined or location. By the way. `getUserCurrentLocation`. It will return either a string or undefined

Something similar to this
`const [locationName] = Object.keys(profileLocationData);`

Reply:
https://github.com/sparkletown/sparkle/pull/1484#discussion_r643612367

Reply:
https://github.com/sparkletown/sparkle/pull/1484/commits/7b918f3a1e72d5576ebab3326949b537d1088345